integrations/stripe/src/index.ts

```diff
@@ -183,14 +183,21 @@ export class Stripe implements TriggerIntegration {
    * });
    * ```
    */
-  onPrice(
-    params?: TriggerParams & { events?: Array<"price.created" | "price.updated" | "price.deleted"> }
-  ) {
-    const event = { ...events.onPrice, name: params?.events ?? events.onPrice.name };
+  onPrice(params?: TriggerParams & { events?: Array<"price.created" | "price.updated" | "price.deleted"> }) {
 
-    return createTrigger(this.source, event, params ?? { connect: false });
+  const validEvents = ["price.created", "price.updated", "price.deleted"];
+
+  const filteredEvents = params?.events?.filter(event => validEvents.includes(event)) ?? events.onPrice.name;
```

Comment:
`filteredEvents` is either a string or string[]

This is bad because on the next line you're checking the length of `filteredEvents`. If it's a string that's going to return the length of the string…